src/v/archival/archival_metadata_stm.cc

```diff
@@ -1356,8 +1378,14 @@ void archival_metadata_stm::apply_add_segment(const segment& segment) {
     bool disable_safe_add
       = config::shard_local_cfg()
           .cloud_storage_disable_metadata_consistency_checks.value();
+
+    auto should_run_optional_validation = !disable_safe_add
+                                          && segment.is_validated
+                                               == segment_validated::yes;
+    auto should_run_mandatory_validation = segment.version > 1;
+
     if (
-      !disable_safe_add && segment.is_validated == segment_validated::yes
+      (should_run_optional_validation || should_run_mandatory_validation)
```

Comment:
that makes sense but I think that the feature flag is required here
the uploads could be just stopped while the cluster is upgrading
or maybe only the v1 should be replicated